lib/GaletteOAuth2/Authorization/UserHelper.php

```diff
@@ -163,36 +170,47 @@ public static function getUserData($container, $id, array $options)
 
         //Add externals groups (free text in info_adh)
         //Example #GROUPS:compta;accueil#
-        if (\preg_match('/#GROUPS:([^#]*([^#]*))#/mui', $member->others_infos_admin, $matches, \PREG_OFFSET_CAPTURE)) {
+        if (preg_match('/#GROUPS:([^#]*([^#]*))#/mui', $member->others_infos_admin, $matches, PREG_OFFSET_CAPTURE)) {
             $g = $matches[1][0];
             Debug::log("Groups added {$g}");
-            $groups = \array_merge($groups, \explode(';', $g));
+            $groups = array_merge($groups, explode(';', $g));
         }
 
         //Reformat group with strtolower, remove whites & slashs
         foreach ($groups as &$group) {
-            $group = \trim($group);
-            $group = \str_replace([' ', '/', '(', ')'], ['_', '', '', ''], $group);
-            $group = \str_replace('__', '_', $group);
-            $group = \mb_strtolower(self::stripAccents($group));
+            $group = trim($group);
+            $group = str_replace([' ', '/', '(', ')'], ['_', '', '', ''], $group);
+            $group = str_replace('__', '_', $group);
+            $group = mb_strtolower(self::stripAccents($group));
+        }
+        $groups = implode(',', $groups);
+
+        $phone = '';
+        if ($member->phone) {
+            $phone = $member->phone;
+        }
+        if ($member->gsm) {
+            if ($phone) {
+                $phone .= '/';
+            }
+            $phone .= $member->gsm;
         }
-        $groups = \implode(',', $groups);
 
         return [
             'id' => $member->id,
             'identifier' => $member->id, //nextcloud
-            'displayName' => $norm_name,
-            'username' => $norm_login,
-            'userName' => $norm_login,
-            'name' => $norm_login,
+            'displayName' => $member->sname,
+            'username' => $norm_login, //FIXME: $member->login,
+            'userName' => $norm_login, //FIXME: $member->login,
+            'name' => $norm_login, //FIXME: $member->sname,
             'email' => $member->email,
             'mail' => $member->email,
             'language' => $member->language,
 
             'country' => $member->country,
             'zip' => $member->zipcode,
             'city' => $member->town,
-            'phone' => $member->phone . '/' . $member->gsm,
+            'phone' => $phone,
```

Comment:
I wonder if it is really useful to send back so much personal information. 

Most of the time, this is totally useless; and when it's not, there is no real way to know if we should send for example "zip", or "zipcode", or even "zipCode"...

Also, having information such as postal address or phone number is quite of normal in Galette, but one may not want to see them on a forum, or other 3rd party services.

I guess restricting personal data send across internet should be the very minimal set.